components/places/src/bookmark_sync/engine.rs

```diff
@@ -490,14 +491,42 @@ fn apply_remote_items(db: &PlacesDb, scope: &SqlInterruptScope, now: Timestamp)
         }
     ).ok(); // We ignore the error because it's expected, users should not have entries with the same id that have different types
 
+    // Same as above but we check if any users have any undetected orphaned bookmarks and
```

Comment:
nit: This comment is a bit confusing because I don't know what "same as above" is referring to. Is it the select statement on `itemsToApply`?